src/lib.rs

```diff
@@ -399,16 +400,20 @@ async fn run(app: AndroidApp) {
             }
             Ok(RuffleEvent::VirtualKeyEvent {
                 down,
-                key_code,
-                key_char,
+                key_descriptor,
             }) => {
                 if let Some(player) = playerbox.as_ref() {
                     let event = if down {
-                        PlayerEvent::KeyDown { key_code, key_char }
+                        PlayerEvent::KeyDown {
+                            key: key_descriptor,
+                        }
                     } else {
-                        PlayerEvent::KeyUp { key_code, key_char }
+                        PlayerEvent::KeyUp {
+                            key: key_descriptor,
+                        }
                     };
                     player.player.lock().unwrap().handle_event(event);
+                    /*
                     if down {
                         // NOTE: this is a HACK
                         if let Some(key) = key_char {
```

Comment:
Do I still need this, @kjarosh?

Reply:
Why did you need this in the first place? :thinking:

Reply:
Just checked, I still do. Because without it, I can't get Robot Unicorn Attack past the second screen, which looks for `on(keyPress "z")`.

Reply:
I suppose winit supplies this event on its own...?

Reply:
Oh, perhaps GH hides it by default... No, it's a `TextInput` event this time.

Reply:
OHHH I'm sorry, I completely missed that. It's not a hack then! That's by design (although non-intuitive). Key down, in order to input text, also has to generate a `TextInput` event, all frontends do that! It's just that in winit you have a dedicated field with text to input, here you reuse the key char.

That's a topic for a different refactor that I'm aiming for, I don't believe text input is different from key char in FP. It's a legacy from the time when KeyCode was identifying a physical key and not a logical key. I have to do some more testing for different keyboard layouts and languages to be sure though.

Reply:
Okay, thanks! I can understand why text input is separate from plain keypress events (for IME reasons and whatnot), but it felt weird to me that the AS `keyPress` even (see sbove) is also triggered by that event, not by the "key down" one. Is that correct?

Reply:
So I don't think TextInput will be useful for IME (it's not used there at all). I think the term "correct" is not well defined. In my opinion there's no need for TextInput, it should either be merged with KeyDown or removed at all.

Reply:
Should `LogicalKey::Character(char)` hold a lowercase or an uppercase letter? Should it be based on shift/capslock? Does it not matter?

Reply:
It should hold a character that is being inputted. So if you press <kbd>Shift</kbd>+<kbd>A</kbd> it will hold `A`, not `a`. If you press <kbd>Alt Gr</kbd>+<kbd>5</kbd> on a Polish keyboard, it will hold `€`.

Reply:
Okay, thanks! So then I especially don't see why TextInput event needs to exist separately.

Reply:
Apparently it can be different on Windows. Additionally, we need to make sure we cover texts for all possible logical keys.

https://docs.rs/winit/latest/winit/event/struct.KeyEvent.html#structfield.text